packages/lab/src/date-input/DateInput.tsx

```diff
@@ -0,0 +1,232 @@
+import { clsx } from "clsx";
+import {
+  ChangeEvent,
+  ComponentPropsWithoutRef,
+  FocusEvent,
+  forwardRef,
+  InputHTMLAttributes,
+  Ref,
+  useState,
+} from "react";
+import { useComponentCssInjection } from "@salt-ds/styles";
+import { useWindow } from "@salt-ds/window";
+
+import dateInputCss from "./DateInput.css";
+import { makePrefixer, useControlled, useFormFieldProps } from "@salt-ds/core";
+import { DateFormatter } from "@internationalized/date";
+
+const withBaseName = makePrefixer("saltDateInput");
+
+const isInvalidDate = (value: string) =>
+  // @ts-ignore evaluating validity of date
+  value && isNaN(new Date(value));
+
+const defaultDateFormatter = (input: string): string => {
+  const date = new Date(input);
+
+  return isInvalidDate(input)
+    ? input
+    : new DateFormatter("EN-GB", {
+        day: "2-digit",
+        month: "short",
+        year: "numeric",
+      }).format(date);
+};
+export interface DateInputProps
+  extends Omit<ComponentPropsWithoutRef<"div">, "defaultValue">,
+    Pick<
+      ComponentPropsWithoutRef<"input">,
+      "disabled" | "value" | "defaultValue" | "placeholder"
+    > {
+  /**
+   * The marker to use in an empty read only DateInput.
+   * Use `''` to disable this feature. Defaults to '—'.
+   */
+  emptyReadOnlyMarker?: string;
+  /**
+   * [Attributes](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/dateInput#Attributes) applied to the `input` elements.
+   */
+  inputProps?: InputHTMLAttributes<HTMLInputElement>;
+  /**
+   * Optional ref for the dateInput component
+   */
+  inputRef?: Ref<HTMLInputElement>;
+  /**
+   * If `true`, the component is read only.
+   */
+  readOnly?: boolean;
+  /**
+   * Alignment of text within container. Defaults to "left"
+   */
+  textAlign?: "left" | "center" | "right";
+  /**
+   * Validation status.
+   */
+  validationStatus?: "error" | "warning" | "success";
+  /**
+   * Styling variant. Defaults to "primary".
+   */
+  variant?: "primary" | "secondary";
+  /**
+   * Function to format the input value.
+   */
+  dateFormatter?: (input: string) => string;
+}
+
+export const DateInput = forwardRef<HTMLDivElement, DateInputProps>(
+  function DateInput(
+    {
+      "aria-activedescendant": ariaActiveDescendant,
+      "aria-expanded": ariaExpanded,
+      "aria-owns": ariaOwns,
+      className: classNameProp,
+      disabled,
+      emptyReadOnlyMarker = "—",
+      inputProps = {},
+      inputRef,
+      readOnly: readOnlyProp,
+      role,
+      style,
+      textAlign = "left",
+      value: valueProp,
+      defaultValue: defaultStartDateProp = valueProp === undefined
+        ? ""
+        : undefined,
+      validationStatus: validationStatusProp,
+      variant = "primary",
+      dateFormatter = defaultDateFormatter,
+      placeholder = "dd mmm yyyy",
+      ...other
```

Comment:
...other instead of ..rest - assuming this is following the same convention as in input, but should we use ..rest for consistency ?